pipeline/pipeline.go

```diff
@@ -132,312 +113,211 @@ func (p *Pipeline) isOutputModule(name string) bool {
 	return found
 }
 
-func GetTraceID(ctx context.Context) (out ttrace.TraceID) {
-	span := ttrace.SpanFromContext(ctx)
-	return span.SpanContext().TraceID()
-}
-
-func (p *Pipeline) computeModuleHashes() {
+func (p *Pipeline) validateAndHashModule(module *pbsubstreams.Module) error {
 	p.moduleHashes = manifest.NewModuleHashes()
+	isOutput := p.outputModuleMap[module.Name]
 
-	for _, module := range p.modules {
-		if p.outputModuleMap[module.Name] {
-			p.moduleHashes.HashModule(p.request.Modules, module, p.graph)
-		}
+	if isOutput && p.reqCtx.StartBlockNum() < module.InitialBlock {
+		return fmt.Errorf("invalid request: start block %d smaller that request outputs for module: %q start block %d", p.reqCtx.StartBlockNum(), module.Name, module.InitialBlock)
 	}
+	p.moduleHashes.HashModule(p.reqCtx.Request().Modules, module, p.graph)
+	return nil
```

Comment:
hmm, this is for a single module now?
Are we not resetting `moduleHashes` at each run then?
That's why we had it handle all the modules in one swift, so that we'd initialize the`NewModulesHashes()` once, and each call to `computeModuleHashes()` would be atomic, initialize and compute anew, in preparation for the time we'd need to recompute the hashes, we'd just call this atomic function and we'll all be set.